BDArmory/Control/BDModuleOrbitalAI.cs

```diff
@@ -0,0 +1,1041 @@
+
+            List<ModuleRCSFX> RCS = v.FindPartModulesImplementing<ModuleRCSFX>();
+            foreach (ModuleRCS thruster in RCS)
+            {
+                if (thruster.useThrottle)
+                    thrust += thruster.thrusterPower;
+            }
+
+            return engines.Sum(e => e.MaxThrustOutputVac(true));
+        }
+        #endregion
+
+        #region Autopilot helper functions
+
+        public override bool CanEngage()
+        {
+            return !vessel.LandedOrSplashed;
+        }
+
+        public override bool IsValidFixedWeaponTarget(Vessel target)
+        {
+            if (!vessel) return false;
+
+            return true;
+        }
+
+        #endregion Autopilot helper functions
+
+        #region WingCommander
+
+        Vector3 GetFormationPosition()
+        {
+            return commandLeader.vessel.CoM + Quaternion.LookRotation(commandLeader.vessel.up, upDir) * this.GetLocalFormationPosition(commandFollowIndex);
+        }
+
+        #endregion WingCommander
+    }
+}
+
+public class BDOrbitalControl : MonoBehaviour //: PartModule
+{
+
+    // /////////////////////////////////////////////////////
+    public Vessel vessel;
+    public Vector3 attitude = Vector3.zero;
+    private Vector3 attitudeLerped;
+    private float error;
+    private float angleLerp;
+    public bool lerpAttitude = true;
+    private float lerpRate;
+    private bool lockAttitude = false;
+
+    private bool facingDesiredRotation;
+    public float throttle;
+    public float throttleActual;
+    internal float throttleLerped;
+    public float throttleLerpRate = 1;
+    public float alignmentToleranceforBurn = 5;
+
+    public Vector3 RCSVector;
+    public float RCSPower = 3f;
+    private Vector3 RCSThrust;
+    private Vector3 up, right, forward;
+    private float RCSThrottle;
+    private Vector3 RCSVectorLerped = Vector3.zero;
+
+    //[KSPEvent(guiActive = true, guiActiveEditor = false, guiName = "ToggleAC")]
+
+    public void Activate()
+    {
+        vessel.OnFlyByWire -= OrbitalControl;
+        vessel.OnFlyByWire += OrbitalControl;
+    }
+
+    public void Deactivate()
+    {
+        vessel.OnFlyByWire -= OrbitalControl;
+    }
+
+    void OrbitalControl(FlightCtrlState s)
+    {
+        error = Vector3.Angle(vessel.ReferenceTransform.up, attitude);
+
+        UpdateSAS(s);
+        UpdateThrottle(s);
+        UpdateRCS(s);
+    }
+
+    private void UpdateThrottle(FlightCtrlState s)
+    {
+        //if (throttle == 0 && throttleLerped == 0) return;
+        //if (v == null) return;
+
+        facingDesiredRotation = error < alignmentToleranceforBurn;
+        throttleActual = facingDesiredRotation ? throttle : 0;
+
+        // Move actual throttle towards throttle target gradually.
+        throttleLerped = Mathf.MoveTowards(throttleLerped, throttleActual, throttleLerpRate * Time.fixedDeltaTime);
+
+        s.mainThrottle = throttleLerped;
```

Comment:
This should probably also set the axis group similarly to `SetThrottle` in BDAirspeedControl.cs.

Similary, `UpdateRCS` should probably also set the `TranslateX/Y/Z` axis groups similarly to `SetFlightControlState` in BDGenericAIBase.cs.